repo/blob/gcs/gcs_storage.go

```diff
@@ -231,30 +209,14 @@ func tokenSourceFromCredentialsJSON(ctx context.Context, data json.RawMessage, s
 func New(ctx context.Context, opt *Options, isCreate bool) (blob.Storage, error) {
 	_ = isCreate
 
-	var ts oauth2.TokenSource
-
-	var err error
-
-	scope := gcsclient.ScopeReadWrite
-	if opt.ReadOnly {
```

Comment:
This seems like a change in behavior as it appear to be dropping read-only scope on the service. I need clarity here.